griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,175 @@
+import uuid
+from typing import Optional
+from attr import define, field, Factory
+from dataclasses import dataclass
+from griptape.drivers import BaseVectorStoreDriver
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine, Column, String, JSON
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Session
+from pgvector.sqlalchemy import Vector
+
+
+@define
+class PgVectorVectorStoreDriver(BaseVectorStoreDriver):
+    """A vector store driver to Postgres using the PGVector extension.
+
+    Attributes:
+        connection_string: An optional string describing the target Postgres database instance.
+        create_engine_params: Additional configuration params passed when creating the database connection.
+        engine: An optional sqlalchemy Postgres engine to use.
+        table_name: Optionally specify the name of the table to used to store vectors.
+    """
+
+    connection_string: str = field(default=None, kw_only=True)
+    create_engine_params: Optional[dict] = field(factory=dict, kw_only=True)
+    engine: Optional[Engine] = field(default=None, kw_only=True)
+    table_name: str = field(default="griptape_vectors", kw_only=True)
+    _model: any = field(default=Factory(lambda self: self.vector_model_factory, takes_self=True), init=False)
+
+    def __attrs_post_init__(self) -> None:
+        """Validates engine and connection string inputs.
+        If a an engine is provided, it will be used to connect to the database. If not, a connection string
+        is used to create a new database connection.
+        """
+        if self.engine is None and self.connection_string is None:
```

Comment:
Can we move this into two validators for `engine` and `connection_string` to make it more explicit?